plugins/cros-ec/fu-cros-ec-usb-device.c

```diff
@@ -675,11 +681,13 @@ fu_cros_ec_usb_device_reset_to_ro(FuCrosEcUsbDevice *self, GError **error)
 						   FALSE,
 						   &error_local)) {
 		/* failure here is ok */
-		g_debug("ignoring failure: %s", error_local->message);
+		g_debug("ignoring failure: reset: %s", error_local->message);
 	}
 
-	/* success */
-	return TRUE;
+	/* immediately disappeared.
+	 * Give some time to detect the device absence
+	 * and prevent sending packets to non-existing USB device. */
+	fu_device_sleep(FU_DEVICE(self), FU_CROS_EC_RESET_DELAY); /* ms */
```

Comment:
This doesn't give me the soft and fuzzies -- what is the device doing here? If it's still "_alive, but unable to process requests_" then I guess this is fine -- if we're delaying where there's a replug happening then it smells like a hack.

Reply:
@hughsie probably you have the other suggestion here.
The device is disappeared here, but `fwupd` immediately switch to the write phase after detach, just like the fwupd doesn't detect as device disappeared (I might be wrong here, I drowned in logs). As far as I see, the first info about the device removed arrived later in about 1-1.5 sec, or even later in rare cases. That's why 2.5 sec delay. Experimentally, I've got at least 1 sec delay on my host, often less than 1.5 sec, rare 2 sec. or more.
The problem then it is switching to detach phase it is still trying to communicate with the non-existing device, having epipe error and aborting the update process. So, it might be considered as a hack.

PS this is a bit strange device -- during the boot it is booted 3 times normally, unless we ask it to stay in RO mode.

Reply:
@hughsie see the attached log of the failed update without the delay 
[update_without_delay.log](https://github.com/user-attachments/files/17356921/update_without_delay.log)
```
failed to write-firmware: failed to transfer block 0x0: failed after 10 retries: failed at sending header: USB error: Pipe error [-9]
```

Reply:
@d4s it smells of a missing `fu_device_add_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG)` somewhere -- i.e. fwupd just needs to know if the device is going to reboot or not. That seems to be the root cause for me -- the early bootloader version expects to reboot and the latter one does not - perhaps?

The reason I'm a bit hesitant to just merge the delay is that the hotplug event is still going to happen -- it's just in a queue of events to be processed. It's much harder to debug what's going on when you see (or try and workaround) the event much after it's happened.

Reply:
@hughsie you are probably right here.
I additionally walk through workflow and looks like here https://github.com/fwupd/fwupd/blob/main/plugins/cros-ec/fu-cros-ec-usb-device.c#L823 should be `FWUPD_ERROR_READ` to catch -EPIPE (have sense to have `FWUPD_ERROR_PIPE` probably?), so flag `FWUPD_DEVICE_FLAG_ANOTHER_WRITE_REQUIRED` is set. And that might be the additioanl proper fix. Testing.

Reply:
> should be FWUPD_ERROR_READ to catch -EPIPE

I think so too!

Reply:
uh, found the correct flow based on private flags and `FWUPD_DEVICE_FLAG_ANOTHER_WRITE_REQUIRED` as it intended to be.
sleep() masked the part of complexity.

@hughsie thanks for pointing!
Now it works without any problems, and even allows restoring the device if it was powered off by occasion in the middle of the flashing :)